avocado/core/loader.py

```diff
@@ -677,7 +681,7 @@ def _make_avocado_tests(self, test_path, make_broken, subtests_filter,
                 test_factories = []
                 for test_class, info in tests.items():
                     if isinstance(test_class, str):
-                        for test_method in info['methods']:
+                        for test_method in sorted(info):
```

Comment:
There's a big problem here.  We don't want the test methods ordered.  We want to keep the same order that was found in the source code.  For the following class:

```python
from avocado import Test

class MyClass(Test):
    def test2(self):
        pass

    def testA(self):
        pass
    
    def test1(self):
        pass

    def testZZZ(self):
        pass
    
    def test(self):
        pass

```

The original behavior is:

```
avocado list foo.py
INSTRUMENTED /tmp/foo.py:MyClass.test2
INSTRUMENTED /tmp/foo.py:MyClass.testA
INSTRUMENTED /tmp/foo.py:MyClass.test1
INSTRUMENTED /tmp/foo.py:MyClass.testZZZ
INSTRUMENTED /tmp/foo.py:MyClass.test
```

With this, it becomes:

```
INSTRUMENTED /tmp/foo.py:MyClass.test
INSTRUMENTED /tmp/foo.py:MyClass.test1
INSTRUMENTED /tmp/foo.py:MyClass.test2
INSTRUMENTED /tmp/foo.py:MyClass.testA
INSTRUMENTED /tmp/foo.py:MyClass.testZZZ
```